### PR TITLE
feat: live reload for playground editor

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
fixes https://github.com/eslint/eslint.org/issues/280

#### What changes did you make? (Give an overview)

Previously build outputs were at `_site` dir where eleventy doesn't detect changes for. We now rather use the actual `asset` folder where the eleventy detects changes and copies files to `_site` folder. Also git ignored the playground and its sourcemap in the asset directory as we dont want that to be commited. 

#### Related Issues

https://github.com/eslint/eslint.org/issues/280

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -3,14 +3,20 @@
 const path = require("path");
 const NodePolyfillPlugin = require("node-polyfill-webpack-plugin");
 
-/** @type {import("webpack").Configuration} */
-module.exports = {
+/**
+ * webpack configuration
+ * @param {import("webpack").ConfigurationFactory} env
+ * @param {import("webpack").ConfigurationFactory} argv
```
Comment (snitin315) [inline webpack.config.js:9]: I don't believe `ConfigurationFactory` is exported by webpack officially. Also, webpack-cli is responsible for these types. https://github.com/webpack/webpack/issues/11630#issuecomment-838510078

```suggestion
 * @param {Record<string, any>} env
 * @param {Record<string, any>} argv
```

**Review by snitin315 — COMMENTED** [review]
🚀 Great fix!

File: webpack.config.js (hunk 1 above)
Comment (harish-sethuraman) [reply webpack.config.js]: Done :)

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**